src/main/java/teammates/common/datatransfer/FeedbackRankOptionsResponseDetails.java

```diff
@@ -65,27 +69,34 @@ public String getAnswerString() {
     public String getAnswerHtml(FeedbackQuestionDetails questionDetails) {
         FeedbackRankOptionsQuestionDetails rankQuestion = (FeedbackRankOptionsQuestionDetails) questionDetails;
         
-        SortedMap<Integer, List<String>> orderedOptions = generateMapOfRanksToOptions(rankQuestion);
-        
+        String invalidAnswer = "Invalid response";
         StringBuilder htmlBuilder = new StringBuilder();
         htmlBuilder.append("<ul>");
         
-        for (Entry<Integer, List<String>> rankAndOption : orderedOptions.entrySet()) {
-            Integer rank = rankAndOption.getKey();
-            if (rank == Const.POINTS_NOT_SUBMITTED) {
-                continue;
-            }
+        if (answers.get(0) == -1) {
+            htmlBuilder.append("<li>");
+            htmlBuilder.append(invalidAnswer);
+            htmlBuilder.append("</li>");
+            
+        } else {
+            SortedMap<Integer, List<String>> orderedOptions = generateMapOfRanksToOptions(rankQuestion);
             
-            List<String> optionsWithGivenRank = rankAndOption.getValue();
-            for (String option : optionsWithGivenRank) {
-                htmlBuilder.append("<li>");
-                htmlBuilder.append(Sanitizer.sanitizeForHtml(rank.toString()));
-                htmlBuilder.append(": ");
-                htmlBuilder.append(option);
-                htmlBuilder.append("</li>");
+            for (Entry<Integer, List<String>> rankAndOption : orderedOptions.entrySet()) {
+                Integer rank = rankAndOption.getKey();
+                if (rank == Const.POINTS_NOT_SUBMITTED) {
+                    continue;
+                }
+                
+                List<String> optionsWithGivenRank = rankAndOption.getValue();
+                for (String option : optionsWithGivenRank) {
+                    htmlBuilder.append("<li>");
+                    htmlBuilder.append(Sanitizer.sanitizeForHtml(rank.toString()));
+                    htmlBuilder.append(": ");
+                    htmlBuilder.append(option);
+                    htmlBuilder.append("</li>");
```

Comment:
Here too. If appending everything together makes it too long, at least append some of them. You don't need to separate every single thing.